services/bls_aggregation/blsagg_test.go

```diff
@@ -0,0 +1,327 @@
+		require.Equal(t, TaskNotFoundErrorF(taskIndex), err)
+	})
+
+	// this is an edge case as typically we would send new tasks and listen for task responses in a for select loop
+	// but this test makes sure the context deadline exceeded can get us out of a deadlock
+	t.Run("send new signedTaskDigest before listening  - task expired", func(t *testing.T) {
+		testOperator1 := types.TestOperator{
+			OperatorId:     types.OperatorId{1},
+			StakePerQuorum: map[types.QuorumNum]types.StakeAmount{0: big.NewInt(100), 1: big.NewInt(200)},
+			BlsKeypair:     newBlsKeyPairPanics("0x1"),
+		}
+		testOperator2 := types.TestOperator{
+			OperatorId:     types.OperatorId{2},
+			StakePerQuorum: map[types.QuorumNum]types.StakeAmount{0: big.NewInt(100), 1: big.NewInt(200)},
+			BlsKeypair:     newBlsKeyPairPanics("0x2"),
+		}
+		blockNum := uint32(1)
+		taskIndex := types.TaskIndex(0)
+		quorumNumbers := []types.QuorumNum{0}
+		quorumThresholdPercentages := []types.QuorumThresholdPercentage{100}
+
+		fakeAvsRegistryService := avsregistry.NewFakeAvsRegistryService(blockNum, []types.TestOperator{testOperator1})
+		noopLogger := logging.NewNoopLogger()
+		blsAggServ := NewBlsAggregatorService(fakeAvsRegistryService, noopLogger)
+
+		err := blsAggServ.InitializeNewTask(taskIndex, blockNum, quorumNumbers, quorumThresholdPercentages, tasksTimeToExpiry)
+		require.Nil(t, err)
+		taskResponseDigest1 := types.TaskResponseDigest{1}
+		blsSigOp1 := testOperator1.BlsKeypair.SignMessage(taskResponseDigest1)
+		err = blsAggServ.ProcessNewSignature(context.Background(), taskIndex, taskResponseDigest1, blsSigOp1, testOperator1.OperatorId)
+		require.Nil(t, err)
+
+		taskResponseDigest2 := types.TaskResponseDigest{2}
+		blsSigOp2 := testOperator2.BlsKeypair.SignMessage(taskResponseDigest2)
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		defer cancel()
+		err = blsAggServ.ProcessNewSignature(ctx, taskIndex, taskResponseDigest2, blsSigOp2, testOperator2.OperatorId)
+		// this should timeout because the task goroutine is blocked on the response channel (since we only listen for it below)
+		require.Equal(t, context.DeadlineExceeded, err)
+
+		wantAggregationServiceResponse := BlsAggregationServiceResponse{
+			Err:                 nil,
+			TaskIndex:           taskIndex,
+			TaskResponseDigest:  taskResponseDigest1,
+			NonSignersPubkeysG1: []*bls.G1Point{},
+			QuorumApksG1:        []*bls.G1Point{testOperator1.BlsKeypair.GetPubKeyG1()},
+			SignersApkG2:        testOperator1.BlsKeypair.GetPubKeyG2(),
+			SignersAggSigG1:     testOperator1.BlsKeypair.SignMessage(taskResponseDigest1),
+		}
+		gotAggregationServiceResponse := <-blsAggServ.ResponseC
+		require.Equal(t, wantAggregationServiceResponse, gotAggregationServiceResponse)
+	})
+
+	t.Run("1 quorum 2 operator 2 signatures on 2 different msgs - task expired", func(t *testing.T) {
```

Comment:
it seems we don't have a test for two different concurrently active tasks

Reply:
Indeed. Need to add this and test with >1 quorums

Reply:
Added them in [cdd35f8](https://github.com/Layr-Labs/eigensdk-go/pull/24/commits/cdd35f8c99748ef2225fbae231581522297e46cb) and [c02f604](https://github.com/Layr-Labs/eigensdk-go/pull/24/commits/c02f6044d56c8b9e4e2d183c3aef9b1a2359e989)

Reply:
I just skimmed over them. Seems good. Resolving conversation.